src/MyLinkedList.java

```diff
@@ -0,0 +1,125 @@
+import org.w3c.dom.Node;
+
+import java.util.LinkedList;
+
+public class MyLinkedList<E> implements List<E> {
+
+    private Node<E> firstNode;
+    private Node<E> lastNode;
+    private int size = 0;
+
+
+    @Override
+    public void add(E value) {
+        Node<E> node = new Node(value, null, null);
+        if (isEmpty()) {
+            firstNode = node;
+            lastNode = node;
+        } else {
+            node.prev = lastNode;
+            lastNode.next = node;
+            lastNode = node;
+        }
+        size++;
+    }
+
+    @Override
+    public void add(E value, int index) {
+        Node<E> node = new Node(value, null, null);
+        if (isEmpty() || index == size) {
+            add(value);
+        } else {
+            Node<E> temp = findIndex(index);
```

Comment:
 does this method work correctly? try to use:
Node<T> temp = findIndex(index);
node.prev = temp.prev;
node.next = temp;
temp.prev.next = node;
temp.prev = node;